src/Utils/multi_region_transformation.jl

```diff
@@ -120,11 +120,9 @@ Base.parent(mo::MultiRegionObject) = construct_regionally(parent, mo)
     multi_region_kwargs = isnothing(findfirst(isregional, kwargs)) ? nothing : kwargs[findfirst(isregional, kwargs)]
     isnothing(multi_region_args) && isnothing(multi_region_kwargs) && return regional_func!(args...; kwargs...)
 
-    if isnothing(multi_region_args)
-        devs = devices(multi_region_kwargs)
-    else
-        devs = devices(multi_region_args)
-    end
+    devs = isnothing(multi_region_args) ? 
+        multi_region_kwargs : multi_region_args |> devices
```

Comment:
How about:

```suggestion
    devs = isnothing(multi_region_args) ? multi_region_kwargs : multi_region_args
    devs = devices(devs)
```

I was not sure if the ternary or pipe operator has precedence but you may know

Reply:
I guess because we can launch `apply_regionally!` with no arguments and only keyword arguments (or with global arguments that do not have devices and regional keyword arguments). By default, we infer the devices we are running on from the arguments, however, in that case, we need to infer the devices from the keyword arguments.

Reply:
Are there important cases where it is not possible to pass arguments? It would simply the implementation to impose the requirement that one of the arguments is a multi region object, so that we do not have to check the kwargs. I feel this might also clarify use of this code as well.